hack/licence-detector/detector/detector.go

```diff
@@ -9,54 +9,66 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"io/ioutil"
 	"path/filepath"
 	"regexp"
 	"strings"
 	"time"
 
+	"github.com/elastic/cloud-on-k8s/hack/licence-detector/dependency"
+	"github.com/google/licenseclassifier"
 	"github.com/karrick/godirwalk"
 )
 
-var errLicenceNotFound = errors.New("failed to detect licence")
+// detectionThreshold is the minimum confidence score required from the licence classifier.
+const detectionThreshold = 0.85
 
-type Dependencies struct {
-	Direct   []LicenceInfo
-	Indirect []LicenceInfo
-}
+var errLicenceNotFound = errors.New("failed to detect licence")
 
-type LicenceInfo struct {
-	Module
-	LicenceFile string
-	Error       error
+type dependencies struct {
+	direct   []*module
+	indirect []*module
 }
 
-type Module struct {
+type module struct {
 	Path     string     // module path
 	Version  string     // module version
 	Main     bool       // is this the main module?
 	Time     *time.Time // time version was created
 	Indirect bool       // is this module only an indirect dependency of main module?
 	Dir      string     // directory holding files for this module, if any
-	Replace  *Module    // replace directive
+	Replace  *module    // replace directive
+}
+
+// MkClassifier creates a new instance of the licence classifier.
+func MkClassifier(dataPath string) (*licenseclassifier.License, error) {
+	absPath, err := filepath.Abs(dataPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine absolute path of licence data file: %w", err)
```

Comment:
nit: it might be worth considering using pkg/errors so we get those sweet, sweet stack traces

Reply:
I think `pkg/errors` is now in maintenance mode in favour of the changes to errors introduced by Go 1.13. I am using the error wrapping method introduced by 1.13 here. I prefer error messages with added context like this over stacktraces as they are easier to read and search for in the code.

Reply:
I think it's been in maintenance mode for a bit now, but you are correct 1.13 adds the error wrapping functionality that pkg/errors had. There's still no easy support afaik for including stack traces as well though. Like I said it was just a personal style suggestion, definitely not a blocker